### ISSUE TITLE
A few use cases: How to center text? Generate page breaks? Does pdf-writer support text wrap?

### ISSUE THREAD

**Issue by Lastofthefirst** [typst/pdf-writer#10]
Hi, I'm poking around the different options for pdf generation in Rust. The use cases I am unsure how to do with pdf-writer are:

* Generate pages while looping through an iterator
* center text
* text wrap

I couldnt figure out how to add new pages so far. I see that there is an enum for centered text, but I didnt see where to pass that in. Lastly, is text wrap supported?

Thanks for the crate, well done. So far I was happy with the generated file sizes.

**Comment by laurmaedje** [comment]
Both PDF and this crate are really low-level, so doing the things you want is not simple. If you just want to create PDFs quickly without having complete control, you might be better off with another crate like [printpdf](https://docs.rs/printpdf/0.5.3/printpdf/). However, it seems, that these also don't really support centered text so far.

To answer your concrete questions:
- To center text, you'd have to manually measure the width of the text and then position each line individually using the `Td` or `Tm` operators. If you're using one of the 14 built-in fonts, you can find the relevant metric (advance width) on the internet. Otherwise, you'd need to extract them from the font you're using.
- To wrap text, it's similar. You have to determine the line breaks yourself using the font and then insert breaks manually.
- To add new pages, you have to add `/Page` objects to the root `/Pages` tree, each with its own content stream.

So in short: You would have to implement the full text layout yourself. Depending on your requirements, this gets arbitrarily complex. Unfortunately, the `TextAlign` enum you saw doesn't do what you want. It's a part of PDF that _describes_ your manual layout in case a mobile PDF viewer wants to reflow the PDF (this feature is almost never used ...)
